src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/fun/notebot/ModuleNotebot.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * This file is part of LiquidBounce (https://github.com/CCBlueX/LiquidBounce)
+ *
+ * Copyright (c) 2015 - 2025 CCBlueX
+ *
+ * LiquidBounce is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LiquidBounce is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.module.modules.`fun`.notebot
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import net.ccbluex.liquidbounce.config.types.Configurable
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.ClientModule
+import net.ccbluex.liquidbounce.features.module.modules.`fun`.notebot.nbs.NbsLoader
+import net.ccbluex.liquidbounce.features.module.modules.`fun`.notebot.nbs.SongData
+import net.ccbluex.liquidbounce.features.module.modules.world.packetmine.ModulePacketMine
+import net.ccbluex.liquidbounce.utils.aiming.RotationsConfigurable
+import net.ccbluex.liquidbounce.utils.client.*
+import net.minecraft.block.enums.NoteBlockInstrument
+import net.minecraft.util.Formatting
+
+/**
+ * Notebot Module
+ *
+ * Automatically plays noteblock songs from NBS files.
+ *
+ * @author ccetl
+ */
+object ModuleNotebot : ClientModule("Notebot", Category.FUN, disableOnQuit = true) {
+
+    private val song by text("SongName", "")
```

Comment:
I recommend using the ItemListValue created through the itemList function to allow users to select songs instead of typing their names. You can create an item with a name and a value, so you can have a normal song name and a file name (Please sanitize it before use to prevent full paths).

In-case you change it to be reloaded on the go, you will need to reload the ClickGUI as well (due to it's cache).

Reply:
I think this input method is kind of easier